src/sentry/incidents/logic.py

```diff
@@ -77,6 +78,41 @@ def create_incident(
     return incident
 
 
+def update_incident_status(incident, status, user=None, comment=None):
+    """
+    Updates the status of an Incident and write an IncidentActivity row to log
+    the change. When the status is CLOSED we also set the date closed to the
+    current time and (todo) take a snapshot of the current incident state.
+    """
+    if incident.status == status.value:
+        # If the status isn't actually changing just no-op.
+        return
```

Comment:
What if they no-op the status but add a comment?

Reply:
I think that if they've passed both then the comment is tied to the status, so I'd probably still not write the comment. Maybe rather than a no-op we could return a validation error.